testsuite/MDAnalysisTests/core/test_atomselections.py

```diff
@@ -812,39 +821,61 @@ def test_nobonds_warns():
 
 
 class TestSelectionErrors(object):
-    def setUp(self):
-        self.u = make_Universe(('names', 'masses',
-                                'resids', 'resnames', 'resnums'))
-
-    def tearDown(self):
-        del self.u
+    @staticmethod
+    @pytest.fixture()
+    def universe():
+        return make_Universe(('names', 'masses', 'resids', 'resnames', 'resnums'))
 
-    def selection_fail(self, selstr):
-        assert_raises(SelectionError, self.u.select_atoms,
+    def selection_fail(self, selstr, universe):
+        assert_raises(SelectionError, universe.select_atoms,
                       selstr)
 
-    def test_expected_errors(self):
-        for selstr in [
-                'name and H',  # string selection
-                'name )',
-                'resid abcd',  # resid arg parsing selection
-                'resnum 7a7',  # rangeselection arg parsing
-                'resid 1-',
-                'prop chicken == tasty',
-                'prop chicken <= 7.4',
-                'prop mass ^^ 12.0',
-                'same this as resid 1',  # same selection
-                'same resid resname mass 5.0',  # same / expect
-                'name H and',  # check all tokens used
-                'naem H',  # unkonwn (misplet) opertaor
-                'resid and name C',  # rangesel not finding vals
-                'resnum ',
-                'bynum or protein',
-                'prop mass < 4.0 hello',  # unused token
-                'prop mass > 10. and group this',  # missing group
-                'prop mass > 10. and fullgroup this',  # missing fullgroup
-        ]:
-            yield self.selection_fail, selstr
+
+    @pytest.mark.parametrize('selstr, universe', [
```

Comment:
I tried that, it didn't work. I don't have a reason to _why_ it didn't work. I'll try to look it up.